core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -7364,18 +7527,30 @@ static class ExtendedExpander extends Expander {
           && (havingExpr
               ? validator.config().conformance().isHavingAlias()
               : !groupByExpr || validator.config().conformance().isGroupByAlias())) {
+
+        // Note: for Groupby/having, aliases in the select list take precedence over
+        // columns in the source table. This is the reverse of the precedence
+        // for where/on clauses.
+
+        // NOTE: this is not True for SF:
```

Comment:
If this is unclear is it possible to throw an exception in the case in may be ambiguous?